File: BackEnd/src/main/java/br/com/projeto2/aajjl/model/Schedule.java
package br.com.projeto2.aajjl.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "Agendamento")
@NoArgsConstructor
@Getter
@Setter
public class Schedule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, unique = true)
    @Setter(AccessLevel.NONE)
    private Long id;

    private Boolean concluido;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "paciente_id", nullable = false)
    private Patient paciente;

    private Shift turno;
    @Column(name = "data_agendamento")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyy-MM-dd")
    private LocalDate dataAgendamento;
    @CreationTimestamp
    @Column(name = "data_criacao", nullable = false, updatable = false)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime dataCriacao;

    private String observacao;
    private String motivoDoAtendimento;
    private Priority prioridade;

    @Override
    public String toString() {
        return "Agendamento{" +
                "concluido=" + concluido +
                ", user=" + user +
                ", paciente=" + paciente +
                ", turno=" + turno +
                ", data=" + dataAgendamento +
                ", observacao='" + observacao + '\'' +
                ", motivoDoAtendimento='" + motivoDoAtendimento + '\'' +
                ", prioridade=" + prioridade +
                '}';
    }
}

File: sistemaFront/src/models/Schedule.ts
export const Shift = {
  MATUTINO : "MATUTINO",
  VESPERTINO : "VESPERTINO",
  NOTURNO : "NOTURNO",
} as const;
export type Shift = typeof Shift[keyof typeof Shift];

export const Priority = {
  VERMELHO : "VERMELHO",
  AMARELO : "AMARELO",
  VERDE : "VERDE",
} as const;
export type Priority = typeof Priority[keyof typeof Priority];

export type ScheduleRequestDTO = {
  userId: number;
  pacienteId: number;
  turno: Shift;
  dataAgendamento: string; // "YYYY-MM-DD"
  observacao: string;
  motivoDoAtendimento: string;
  prioridade: Priority;
};

export type ScheduleUpdateDTO = {
  concluido?: boolean;
  turno?: Shift;
  dataAgendamento?: string;
  observacao?: string;
  motivoDoAtendimento?: string;
  prioridade?: Priority;
};

type UserInfoDTO = {
  id: number;
  nome: string;
  profissao: string;
};

type PatientInfoDTO = {
  id: number;
  nome: string;
};

export type ScheduleResponseDTO = {
  id: number;
  concluido: boolean;
  user: UserInfoDTO;
  paciente: PatientInfoDTO;
  turno: Shift;
  dataAgendamento: string;
  dataCriacao: string;
  observacao: string;
  motivoDoAtendimento: string;
  prioridade: Priority;
};